packages/webpack/src/plugin.js

```diff
@@ -248,34 +279,23 @@ class LavaMoatPlugin {
           Array.from(chunks).forEach((chunk) => {
             chunkGraph.getChunkModules(chunk).forEach((module) => {
               const moduleId = chunkGraph.getModuleId(module)
-              // TODO: potentially a place to hook into for policy generation
-              // Module policies can be merged into a per-package policy in a second pass or right away, depending on what we can pull off for the ID generation.
-              // Seems like it'd make sense to do it right away.
-              if (isNormalModule(module)) {
-                policyGenerator.inspectWebpackModule(
-                  module,
-                  compilation.moduleGraph.getOutgoingConnections(module)
-                )
-              }
+              const moduleClass = Object.getPrototypeOf(module).constructor.name
 
-              if (
-                // Webpack has a concept of ignored modules
-                // When a module is ignored a carveout is necessary in policy enforcement for it because the ID that webpack creates for it is not exactly helpful.
-                // example outcome in the bundle: `const nodeCrypto = __webpack_require__(/*! crypto */ "?0b7d");`
-                // Sadly, even treeshaking doesn't eliminate that module. It's left there and failing to work when reached by runtime policy enforcement.
-                // Below is the most reliable way I've found to date to identify ignored modules.
-                (module.type === JAVASCRIPT_MODULE_TYPE_DYNAMIC &&
-                  // @ts-expect-error BAD TYPES
-                  module.identifierStr?.startsWith('ignored')) ||
-                // @ts-expect-error BAD TYPES
-                module.resource === undefined // better to explicitly list it as unenforceable than let it fall through the cracks
-              ) {
+              if (isIgnoredModule(module)) {
                 unenforceableModuleIds.push(moduleId)
               } else {
+                if (isInspectableModule(module, moduleClass)) {
+                  policyGenerator.inspectWebpackModule(
+                    module,
+                    compilation.moduleGraph.getOutgoingConnections(module)
+                  )
+                }
+
+                // typescript is complaining about the use of `resource` here, but it's actually there.
```

Comment:
would be more appropriate to use whatever the actual type is, instead of `any`, but nbd

Reply:
There's a bunch of types and non-breaking changes to webpack may add more and there's no good type they all inherit from. the base Module type doesn't have the field. Almost all other SomethingModule types do.